src/domainobjectfactories/back_office_position_factory.py

```diff
@@ -45,30 +43,107 @@ def create_record(self):
             A single back office position object
         """
 
-        instrument = self.get_random_instrument()
-        position_type = self.create_position_type()
-        knowledge_date = self.create_knowledge_date()
+        instrument_id, isin = self.__create_instrument_details()
+        account_id, account_type = self.__create_account_details()
+
         record = {
-            'cusip': instrument['cusip'],
-            'position_type': position_type,
-            'knowledge_date': knowledge_date,
-            'effective_date': self.create_effective_date(
-                                2, knowledge_date, position_type),
-            'account': self.create_account(
-                account_types=self.BACK_OFFICE_ACCOUNT_TYPES
-                ),
-            'direction': self.create_credit_debit(),
-            'qty': self.create_random_integer(),
-            'purpose': self.create_purpose(),
-            'time_stamp': datetime.now(),
+            'as_of_date': self.__create_as_of_date(),
+            'value_date': self.__create_value_date(),
+            'ledger': self.__create_ledger(),
+            'instrument_id': instrument_id,
+            'isin': isin,
+            'account_id': account_id,
+            'account_type': account_type,
+            'quantity': self.__create_quantity(),
+            'purpose': self.__create_purpose()
         }
 
         for key, value in self.create_dummy_field_generator():
             record[key] = value
 
         return record
 
-    def create_purpose(self):
+    @staticmethod
+    def __create_as_of_date():
+        """ Return the 'as of date', which must be the current date
+        Returns
+        -------
+        Date
+            Date object representing the current date
+        """
+        return datetime.date.today()
```

Comment:
I've raised #192 so we know to fix the date and datetime generation across all objects but we may as well do this correctly for back office positions first time around.  I think the best way we found to generate a date using UTC time is:
datetime.datetime.now(datetime.timezone.utc).date()